### ISSUE TITLE
[Important] [Windows] Distortion of window while starting up

### ISSUE THREAD

**Issue by KRTirtho** [leanflutter/window_manager#464]
After upgrading to **Flutter 3.22.1**, Window manager is causing the entire UI to be glitched out (distorted) initially after launch.

This is how it looks
![image](https://github.com/leanflutter/window_manager/assets/61944859/1fefa699-6dc4-46ea-bfa9-2ef12c1c1383)

If I turn off/comment out `windowManager.ensureInitialized` and `windowManager.waitUntilReadyToShow` function calls, the app shows up completely fine except the Title bar is visible (as expected).

There are no errors/exceptions being thrown at the Console. So, I guess this is caused any embedder change in Flutter engine?

> Reverting back to Flutter 3.19.x fixes the issue though

Upstream issue: https://github.com/KRTirtho/spotube/issues/1553

Thanks for your amazing work and the plugin @lijy91 ❤️ 
Please fix this as soon as possible. Lots of users are affected by it.

**Comment by KRTirtho** [comment]
There were lots of changes in the Windows embeder EGL which probably lead to this kind of issues.
Special focus: https://docs.flutter.dev/release/release-notes/release-notes-3.22.0#desktop

**Comment by Linij** [comment]
![image](https://github.com/user-attachments/assets/470a12fc-e93d-4b92-b8e7-cfa2b0f5d7ed)
![image](https://github.com/user-attachments/assets/38c0e6ad-c5e7-4147-a9a0-c06292b01d3c)
I encountered the same problem, but I found a solution that requires modifying the size in the main.cpp default file to be the same as the set minimum value to fix it。

**Comment by KRTirtho** [comment]
I tried your fix but the issue comes back after opening the app 2/3 times.

**Comment by KRTirtho** [comment]
There's a workaround suggested by [Far Se on Flutter Discord](https://discord.com/channels/420324994703163402/570319299177218049/1266318868310392875) that allows to use Flutter v3.22.x with window_manager in Windows

```dart
    WidgetsBinding.instance.addPostFrameCallback((Duration timeStamp) async {
      if (Theme.of(context).platform != TargetPlatform.windows) return;
      await Future<void>.delayed(const Duration(milliseconds: 100), () {
        windowManager.getSize().then((Size value) {
          windowManager.setSize(
            Size(value.width + 1, value.height + 1),
          );
        });
      });
    });
```

Add following in the root App's (the very first widget) `initState` method

**Comment by Linij** [comment]
OK，I will try this. Thinks.

**Comment by Linij** [comment]
This method is feasible, but I think it may be better。
WidgetsBinding.instance.addPostFrameCallback((Duration timeStamp) async {
      if (Theme.of(context).platform != TargetPlatform.windows) return;
      await Future<void>.delayed(const Duration(milliseconds: 100), () {
        windowManager.getSize().then((Size value) {
          windowManager.setSize(
            Size(value.width + 1, value.height + 1),
          );
          windowManager.setSize(
            Size(value.width, value.height),
          );
        });
      });
    });

**Comment by maznnwell** [comment]
I also encountered this issue, but the suggested solution didn't worked for me - on Flutter 3.24.3.

Somewhere in the widget tree, after creating the MaterialApp, a Lottie was being played using a controller - but it was rendering only when it was done loading from the asset file. So I decided to have something rendering while the lottie was loading and it worked.

**Comment by KRTirtho** [comment]
hey @lijy91 if you can look into this issue when you get time? Lots of users are suffering for this.
I'm also not finding any clue why this might happen.

**Comment by leeys0207** [comment]
I found a hacky solution. 
await windowManager.maximize();
await windowManager.unmaximize();

**Comment by Xazin** [comment]
We also have this issue in AppFlowy, we ended up switching to bitsdojo on windows rather than window_manager.

I want to switch back, but the workarounds are hacky _(and not usable for all cases)_ and ends up with a bad user experience.

**Comment by KRTirtho** [comment]
On Spotube's case, both `window_manager`and `bitsdojo_window` are having the same issue. How well does `bitsdojo_window` work for AppFlowy?

**Comment by Xazin** [comment]
> On Spotube's case, both `window_manager`and `bitsdojo_window` are having the same issue. How well does `bitsdojo_window` work for AppFlowy?

It has the same issue, but seems to happen less frequently, not really anything I can prove though.

We prefer window_manager though.

**Comment by artdedeco** [comment]
Did anyone ever find a way to at least consistently repro this?

**Comment by Far-Se** [comment]
@artdedeco yes, if you do not "draw"/use a widget from pixel [0,X], as in from the beginning of the app, it will render like that. I have two "hacks for this":
1. Size adjustement in initState:
```dart
    WidgetsBinding.instance.addPostFrameCallback((Duration timeStamp) async {
      await Future<void>.delayed(const Duration(milliseconds: 100), () async {
        final Size value = await windowManager.getSize();
        await windowManager.setSize(Size(value.width + 1, value.height + 1));
        await windowManager.setSize(Size(value.width, value.height));
      });
    });
```
2. Draw a barely visible line on top of the app:
```dart
              MaterialApp(
                  scrollBehavior: MyCustomScrollBehavior(),
                  debugShowCheckedModeBanner: false,
                  title: 'Tabame',
                  theme: AppTheme.getLightThemeData(),
                  darkTheme: AppTheme.getDarkThemeData(context),
                  themeMode: themeMode,
                  home: Stack(
                          children: <Widget>[
                            Positioned(
                              top: 0,
                              left: 0,
                              child: Container(width: 100, height: 1, color: Colors.black.withValues(alpha: 0.001)),
                            ),
                            Material(
                                type: MaterialType.transparency,
                                child: QuickMenu(),
                          ],
                        ))
```

The second one seems to work better.

Edit: the second one does not work. Really bad ugly bug from Flutter side.

**Comment by damywise** [comment]
Apparently this will be fixed in https://github.com/flutter/flutter/pull/187954

**Comment by Far-Se** [comment]
<img width="1608" height="823" alt="Image" src="https://github.com/user-attachments/assets/f26220ec-d42c-4733-b702-93af66134a24" />
Only 2 lines changed to fix a bug that has annoyed me for years 😂

**Comment by lijy91** [comment]
Closing: this was a Flutter engine bug (window size transposed in OnEmptyFrameGenerated), fixed upstream in flutter/flutter#187954 and included in Flutter 3.47 stable. Please upgrade Flutter; if you still see the distortion on 3.47 or later, feel free to open a new issue with a reproduction.
